### PR TITLE
Terraform support

### PR DESCRIPTION
**Pull request by derBroBro**
### What does this implement/fix? Explain your changes.
This adds support for terraform fmt. 
### Does this close any currently open issues?
Yes - https://github.com/Glavin001/atom-beautify/issues/645

### Any other comments?
Hope all fits.
```
npm run postinstall
```
Did not work.
As I am no fulltime dev it would be great if you can guide me what else I have to deliver.
### Checklist

Check all those that are applicable and complete.

- [x] Merged with latest `master` branch
- [x] Regenerate documentation with `npm run docs`
- [x] Add change details to `CHANGELOG.md` under "Next" section
- [x] Added examples for testing to [examples/ directory](examples/)
- [x] Travis CI passes (Mac support)
- [x] AppVeyor passes (Windows support)

### CODE REVIEW

**Comment by derBroBro** [comment]
Travis failed because:
```
/usr/local/Homebrew/Library/Homebrew/brew.rb:12:in `<main>': Homebrew must be run under Ruby 2.3! (RuntimeError)
```
Have I to fix something or is this a pipeline thing?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Next
+- See [](). Add support for Terraform fmt.
```
Comment (Glavin001) [inline CHANGELOG.md]: Please fix the link:

```
See [#645](https://github.com/Glavin001/atom-beautify/issues/645). Add support for Terraform fmt.
```

File: src/beautifiers/terraformfmt.coffee [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+Requires terraform installed
+###
+
+"use strict"
+Beautifier = require('./beautifier')
+
+module.exports = class Terraformfmt extends Beautifier
+  name: "terraformfmt"
+  link: "https://www.terraform.io/docs/commands/fmt.html"
+  isPreInstalled: false
```
Comment (Glavin001) [inline src/beautifiers/terraformfmt.coffee]: New beautifiers should use `executables` feature: https://github.com/Glavin001/atom-beautify/issues/1687

Here's one Pull Request for example: https://github.com/Glavin001/atom-beautify/pull/1726/files#diff-4bf70ccc1cd2e9c4d2706b85b61fb41c

Please let me know if you have any questions. The Executables feature needs more documentation, so currently all we have are existing beautifiers are examples. See https://github.com/Glavin001/atom-beautify#beautifiers

File: src/beautifiers/terraformfmt.coffee [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  options: {
+    Terraform: true
+  }
+
+  beautify: (text, language, options) ->
+    @run("terraform", [
```
Comment (Glavin001) [inline src/beautifiers/terraformfmt.coffee]: Use Executables feature.

```
# Before
@run("terraform", [

# After
@exe("terraform").run([
```

File: src/beautifiers/terraformfmt.coffee [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+"use strict"
+Beautifier = require('./beautifier')
+
+module.exports = class Terraformfmt extends Beautifier
+  name: "terraformfmt"
+  link: "https://www.terraform.io/docs/commands/fmt.html"
+  isPreInstalled: false
+
+  options: {
+    Terraform: true
```
Comment (Glavin001) [inline src/beautifiers/terraformfmt.coffee]: Should be `false` indicating Language is supported *with no/zero support for options*.

File: src/options.json [hunk 5]
```diff
@@ -9128,6 +9128,340 @@
...
+      "terraformfmt"
+    ],
+    "grammars": [
+      "Terraform"
+    ],
+    "extensions": [
+      "tf"
+    ],
+    "properties": {
+      "indent_size": {
```
Comment (Glavin001) [inline src/options.json]: Need to fix `Terraform: true` above and rerun generate docs.

**Comment by Glavin001** [comment]
> npm run postinstall
> Did not work.

I don't think there is a `postinstall` command: https://github.com/Glavin001/atom-beautify/blob/master/package.json#L412-L419

I think you meant `npm run docs`. What was the error?

**Comment by derBroBro** [comment]
For the question regarding postinstall I used this: https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md

File: src/beautifiers/terraformfmt.coffee (hunk 2 above)
Comment (derBroBro) [reply src/beautifiers/terraformfmt.coffee]: If I change this, the exec is not found anymore. Testing with where (on windows) works.
Any Idea?

Comment (derBroBro) [reply src/beautifiers/terraformfmt.coffee]: @Glavin001 now everything should be fixed.

**Comment by derBroBro** [comment]
I am not sure what I have to change to fix the issues. Can you support here ?

**Comment by szeck87** [comment]
Hi @derBroBro you need to have Terraform installed on both the AppVeyor (WIndows) and Travis CI (Linux and MacOS) build files for the tests to pass.  .travis.yml for Travis and appveyor.yml for AppVeyor.

For Travis, you can use [uncrustify](https://github.com/Glavin001/atom-beautify/blob/master/.travis.yml#L126-L131) as an example, as it looks like there is a Homebrew formula for Terraform:  http://formulae.brew.sh/formula/terraform.  Basically if the OS is OSX, `brew install terraform` otherwise `docker pull <docker image here`.

**Comment by derBroBro** [comment]
Thank you, now appveyor looks ok, travis throws ```/usr/local/Homebrew/Library/Homebrew/brew.rb:12:in `<main>': Homebrew must be run under Ruby 2.3! (RuntimeError)```

**Comment by szeck87** [comment]
There's an issue with Homebrew on Travis, uncomment line 95: https://github.com/Glavin001/atom-beautify/blob/master/.travis.yml#L95 and push it again.

**Comment by derBroBro** [comment]
Worked, I think with this the I changed everything. 
The changes requested should be also fulfilled (not sure why not detected)

**Comment by derBroBro** [comment]
@Glavin001 anything I can to finish the PR?

**Comment by derBroBro** [comment]
Hello, any updates here?

**Comment by szeck87** [comment]
Looks good to me, I requested another review from Glavin.

**Comment by Glavin001** [comment]
Reviewing now. I apologize, I have been unavailable to review Pull Requests lately. Let's see about getting this merged 😄 .

File: src/beautifiers/terraformfmt.coffee [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+Requires terraform installed
+###
+
+"use strict"
+Beautifier = require('./beautifier')
+
+module.exports = class Terraformfmt extends Beautifier
+  name: "terraformfmt"
+  link: "https://www.terraform.io/docs/commands/fmt.html"
+  isPreInstalled: false
```
Comment (Glavin001) [inline src/beautifiers/terraformfmt.coffee]: `isPreInstalled` should be removed. See https://github.com/Glavin001/atom-beautify/blob/49982c6ce7b0c8f66042c2eddbc49d1441944023/src/beautifiers/beautifier.coffee#L40-L41

**Comment by Glavin001** [comment]
Looks like merge conflicts now, too. Once resolved, should be good to go!

**Comment by Glavin001** [comment]
Awesome! Thank you for contributing!

**Comment by Glavin001** [comment]
Published to v0.30.6

**Pull request merged**